app/jobs/data_exports/create_job.rb

```diff
@@ -256,5 +259,20 @@ def map_metadata_fields(metadata_fields, sample_metadata)
         end
       end
     end
+
+    # Queries---------------------------------------------------------------------
+    def sample_query(data_export)
+      Sample.includes(project: :namespace, attachments: { file_attachment: :blob })
+            .where(id: data_export.export_parameters['ids'])
+    end
+
+    def attachments_query(sample, data_export)
+      if data_export.export_parameters.key?('attachment_formats')
+        sample.attachments.where("metadata->'format' ?| array[:formats]",
```

Comment:
**nitpick (blocking):** Since we use `includes` above and specify `attachments`, I think to reduce extra queries here you could just filter the collection instead of performing an additional query. I have not watch the logs to see the effect of this when the job is running.